src/auth/authenticator.rs

```diff
@@ -95,6 +99,21 @@ impl From<&str> for Credentials {
 #[derive(Clone, Eq, PartialEq)]
 pub struct ClientCert(pub Vec<u8>);
 
+use x509_parser::prelude::parse_x509_certificate;
+
+impl ClientCert {
+    /// Returns true if the Common Name from the client certificate matches the allowed_cn
+    pub fn verify_cn(&self, allowed_cn: &str) -> Result<bool, std::io::Error> {
+        let client_cert = parse_x509_certificate(&self.0);
+        let subject = match client_cert {
+            Ok(c) => c.1.subject().to_string(),
+            Err(e) => return Err(std::io::Error::new(std::io::ErrorKind::Other, e.to_string())),
+        };
+
+        Ok(subject.contains(allowed_cn))
```

Comment:
should it be contains or equals?

Reply:
contains because the subject holds the DN